src/thunderbots/software/grsim_communication/grsim_command_primitive_visitor.cpp

```diff
@@ -153,7 +153,12 @@ void GrsimCommandPrimitiveVisitor::visit(const PivotPrimitive &pivot_primitive)
 
 void GrsimCommandPrimitiveVisitor::visit(const DribblePrimitive &dribble_primitive)
 {
-    // TODO: https://github.com/UBC-Thunderbots/Software/issues/107
+    float kickPower =
+        (float)(dribble_primitive.isSmallKickAllowed() ? DRIBBLE_SMALL_KICK_POWER : 0.0);
+
+    motion_controller_command = MotionController::MotionControllerCommand(
+        dribble_primitive.getDestination(), dribble_primitive.getFinalAngle(),
+        dribble_primitive.getFinalSpeed(), kickPower, false, true);
```

Comment:
The way `kickPower` is being used here, the robot will always kick the ball when it comes into contact with it, which isn't quite right since then we won't spend time dribbling. We only need to do a small kick before we have moved 1 meter from the starting location of our dribble, so we don't violate the rules.

(The rules are that the robot cannot dribble the ball for more than 1 meter without releasing the ball)

Reply:
After some offline discussion we are leaving this for now as it's not a priority, and wouldn't be simulated well anyway.